jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/OpenJ9DiagnosticsMXBeanImpl.java

```diff
@@ -0,0 +1,164 @@
+/*[INCLUDE-IF Sidecar17]*/
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+
+package com.ibm.lang.management.internal;
+
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+
+import com.ibm.jvm.Dump;
+import com.ibm.jvm.DumpPermission;
+import com.ibm.jvm.DumpConfigurationUnavailableException;
+import com.ibm.jvm.InvalidDumpOptionException;
+
+import com.ibm.jvm.Trace;
+import com.ibm.lang.management.OpenJ9DiagnosticsMXBean;
+
+public class OpenJ9DiagnosticsMXBeanImpl implements OpenJ9DiagnosticsMXBean {
+	private static final DumpPermission DUMP_PERMISSION = new DumpPermission();
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void resetDumpOption() throws DumpConfigurationUnavailableException {
+		Dump.resetDumpOptions();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void setDumpOptions(String dumpOptions) throws InvalidDumpOptionException, DumpConfigurationUnavailableException {
+		Dump.setDumpOptions(dumpOptions);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void triggerDump(String dumpAgent) throws IllegalArgumentException {
+		/* Check the caller is allowed to trigger a dump. */
+		checkDumpSecurityPermssion();
+                switch (dumpAgent) {
+                        case "java":
+                                Dump.JavaDump();
+				break;
+                        case "heap":
+                                Dump.HeapDump();
+				break;
+                        case "system":
+                                Dump.SystemDump();
+				break;
+                        case "snap":
+                                Dump.SnapDump();
+				break;
+                        default:
+                                throw new IllegalArgumentException("Invalid or Unsupported Dump Agent Option, cannot be triggered");
```

Comment:
Please externalize this message. For an example, search for `/*[MSG` and also see openj9/jcl/src/java.base/share/classes/com/ibm/oti/util/ExternalMessages-MasterIndex.properties